File: cluster/midonet-cluster/src/main/java/org/midonet/cluster/rest_api/models/federation/MidonetVtep.java
/*
 * Copyright 2015 Midokura SARL
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.midonet.cluster.rest_api.models.federation;

import java.net.URI;
import java.util.List;
import java.util.UUID;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonIgnore;

import org.midonet.cluster.data.ZoomClass;
import org.midonet.cluster.data.ZoomField;
import org.midonet.cluster.models.Federation;
import org.midonet.cluster.rest_api.models.UriResource;
import org.midonet.cluster.rest_api.validation.MessageProperty;
import org.midonet.cluster.util.IPAddressUtil;
import org.midonet.cluster.util.UUIDUtil.Converter;
import org.midonet.packets.IPv4;

@ZoomClass(clazz = Federation.MidoNetVtep.class)
public class MidonetVtep extends UriResource {

    @ZoomField(name = "id", converter = Converter.class)
    public UUID id;

    @ZoomField(name = "name")
    public String name;

    @ZoomField(name = "username")
    public String username;

    @ZoomField(name = "password")
    public String password;

    @ZoomField(name = "group_id", converter = Converter.class)
    public UUID groupId;

    @ZoomField(name = "vtep_router_id", converter = Converter.class)
    public UUID vtepRouterId;

    @Size(min = 1)
    @Pattern(regexp = IPv4.regex, message = MessageProperty.IP_ADDR_INVALID)
    @ZoomField(name = "network_api_endpoint", converter = IPAddressUtil.Converter.class)
    public List<String> networkApiEndpoint;

    public MidonetVtep() {
    }

    @Override
    public URI getUri() {
        return absoluteUri(Application.MIDONET_VTEP, id);
    }

    @JsonIgnore
    @Override
    public void create() {
        if (null == id) {
            id = UUID.randomUUID();
        }
    }

    @JsonIgnore
    public void update(MidonetVtep from) {
        this.id = from.id;
        vtepRouterId = from.vtepRouterId;
        groupId = from.groupId;
    }
}

File: python-midonetclient/src/midonetclient/federation/midonet_vtep.py
# vim: tabstop=4 shiftwidth=4 softtabstop=4

# Copyright 2015 Midokura SARL
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

import logging

from midonetclient import resource_base
from midonetclient.federation import vendor_media_type

LOG = logging.getLogger(__name__)


class MidonetVtep(resource_base.ResourceBase):
    media_type = vendor_media_type.MIDONET_VTEP_JSON

    def __init__(self, uri, dto, auth):
        super(MidonetVtep, self).__init__(uri, dto, auth)

    def get_id(self):
        return self.dto['id']

    def get_name(self):
        return self.dto['name']

    def name(self, name):
        self.dto['name'] = name
        return self

    def get_group(self):
        return self.dto['group']

    def group(self, group):
        self.dto['group'] = group
        return self

    def get_vtep_router(self):
        return self.dto['vtepRouter']

    def vtep_router(self, vtepRouter):
        self.dto['vtepRouter'] = vtepRouter
        return self

    def get_username(self):
        return self.dto['username']

    def username(self, username):
        self.dto['username'] = username
        return self

    def get_password(self):
        return self.dto['password']

    def password(self, password):
        self.dto['password'] = password
        return self

    def get_api_endpoints(self):
        return self.dto['networkApiEndpoint']

    def api_endpoints(self, apiEndpoint):
        self.dto['networkApiEndpoint'] = apiEndpoint
        return self
